File: FindMinimuminRotatedSortedArray.py
class Solution:
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        left, right = 0, len(nums)-1
        while left < right:
            if nums[left] < nums[right]:
                return nums[left]
            mid = (left+right)//2
            if nums[mid] >= nums[left]:
                left = mid+1
            else:
                right = mid
        return nums[left]
        
if __name__=='__main__':
    nums = list(input())
    solution = Solution()
    print(solution.findMin(nums))

File: FindMinimuminRotatedSortedArray.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    int findMin(vector<int>& nums) {
        int min = nums[0];
        for(int i=0;i<nums.size()-1;i++){
            if(nums[i]>nums[i+1]){
                min = nums[i+1];
                break;
            }
        }
        
        return min;
    }
};

int main(){
    
    int n;
    cin>>n;
    
    vector<int> nums;
    for(int i=0;i<n;i++){
        int num;
        cin>>num;
        nums.push_back(num);
    }
    
    Solution *solution = new Solution();
    cout<<solution->findMin(nums);
    
    return 0;
}
